keps/sig-windows/1981-windows-privileged-container-support/README.md

```diff
@@ -459,87 +464,136 @@ Windows privileged containers will be implemented with [Job Objects](https://doc
+- Directory mounts (i.e. directory C:\test mapped to C:\test in the privileged container) are still being investigated. Note that file system mapping in the current implementation would expose both the host filesystem and the container filesystem to the processes in the privileged container. We are still investigation how external mounts such as secrets or storage accounts would be added. The default visibility of all the host filesystem may present backwards compatibility challenges in the future if host path mounting is enabled in Windows. Currently the only method to restrict would be via the user in a way similar to [UID or GID](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) in the PSP. Currently however, this host path mounting is unlikely to be enabled in the future as a restriction of Job Objects.
+
 #### Container Images
-- A slim base image may be required to satisfy hierarchy requirements from HCS. It was found that the graphdriver calls expect certain files to be in a windows image when un-tarring. These files were found to be simply several registry hive files in /windows/system32/config (which can be empty).
-- We are currently targeting to release a simple image alongside this feature including the above mentioned files, which can be used instead of traditional Windows base images of server core or otherwise. Server core and others that container the necessary files would also work as privileged container base images. 
+
+- Privileged containers can be built on top of existing Windows base images (nanoserver, servercore, etc).
+- A slim base image may be introduced to act as a replacement for [scratch]. On windows graphdriver calls expect some files (mainly registry hives) and these would be included in the slim base image.
+- Privileged containers will not inherit the same [compatibility requirements](https://docs.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/version-compatibility) as process isolated containers from an OS perspective. Container runtimes like containerd may need to skip OS version checks when pulling/starting these containers. This will continue to be investigated as the feature matures.
+
 #### Container Image Build/Definition
-- This is another ongoing area of investigation and open to feedback. 
- 
+
```

Comment:
From the "Open Questions", it seems that the image built will only be able to run as privileged?

If that is the case, can we add some information on the image, so that the container runtime can check that and report clear error? Or else, users may struggle to figure out what is wrong.

Reply:
What will happen, and what will users see when they try to run a regular windows container image as host job, or host job windows container image as regular container?

It is not very clear currently in this doc.

Reply:
I don't believe setting registry values in the trimmed down image will work or show up on the host. And @Random-Liu correct, this new image will only function for (just going to go with host job here) host job containers. It's missing the core Windows pieces to be able to boot from for normal containers (the trimmed image is essentially empty, almost akin to FROM scratch on Linux).

Reply:
> I don't believe setting registry values in the trimmed down image will work or show up on the host.

I believe the image manifest / config does carry information, e.g. annotations. It should be possible to inject some information during image build, so that when the container runtime runs the container, it can get that information from the image manifest / config.

This is just some idea, :p But the more important thing is to make sure users can quickly identify the issue (or we can even prevent it in the future), if they there is a mismatch.

For windows, the file system in the host job container image and regular container image are quite different, which makes it a bit unclear whether users can get a reasonable error when they have a mismatch.

Another question I have is that, for a container image, before a container can use it, in containerd we usually go through `download content`, `unpack into snapshot`, `create container rootfs`. For this "host job" container image, does it only have logic difference in the last step?

Reply:
These are all great questions.
As stated in various places in KEP we've got a lot of open questions well need to investigate and a whole lot of documentation we'll need to do.
In my opinion these questions shouldn't block started implementation towards an alpha state.
It would be nice to get the CRI changes in soon so we can also start the work needed in containerd.

Reply:
@marosset I think most of my concerns have been addressed, these are just some non-blocking questions.

I thought this KEP is waiting for other approvals. I can give my LGTM after a final round of review today.

Reply:
Yup, I'm waiting on feedback for the API changes and the PRR review.

Reply:
BTW, for the API change, still I'm fine with both reusing the current privileged field, or adding a new field with more descriptive name.

Let's see what people's option from the API review.

Reply:
I brought this up at the Feb 3rd SIG-Auth meeting and the feeling I got from the discussion was people thought it better to not-reuse privileged if it didn't match existing / Linux behaviors. Still waiting for their comments here tho.

@Random-Liu the other thing of note was the suggestion to create a `WindowsPod` API object to better represent Windows behaviors for pod's on k8s. I'm going to try and put together a proposal for this when I get some time
Would do you think about that?

Reply:
> the other thing of note was the suggestion to create a WindowsPod API object to better represent Windows behaviors for pod's on k8s.

Do you mean defining a `WindowsPod` API object to replace the current `Pod` API?

Hm, that sounds like a bigger API change, and something we should do before windows support is GA. Since now windows support has already been GA, and people have started using it. It is not clear to me how we can easily switch the API to a separate `WindowsPod` object without breaking the API backward compatibility. That would probably add even more churns and maintenance overhead.

At least, that sounds like a separate effort to me.

Reply:
`WindowsPod` as a peer to `Pod` objects would **definitely** be a sperate effort. 
I was mentioning this here just because it could help solve a lot of issues we have trying to describe Windows functionality on Pods\Specs that were originally built with only Linux in mind :)